bindings/c/test/mako/mako.cpp

```diff
@@ -581,70 +603,76 @@ int runWorkload(Database db,
 	/* main transaction loop */
 	while (1) {
 		Transaction tx = createNewTransaction(db, args, -1, args.active_tenants > 0 ? tenants : nullptr);
-		while ((thread_tps > 0) && (xacts >= current_tps)) {
+		if ((thread_tps > 0 /* iff throttling on */) && (xacts >= current_tps)) {
```

Comment:
As it is we're building up to a transaction timeout while we wait for TPS to be relaxed again.
It makes more sense to move the `createNewTransaction()` below the throttle block.